src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] array = new Object[10];
+    private int size = 1;
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            Object[] newArray = new Object[array.length + array.length / 2];
+            System.arraycopy(array, 0, newArray, 0, array.length);
+            array = newArray;
+        }
+        array[size - 1] = value;
```

Comment:
The assignment `array[size - 1] = value;` is incorrect. It should be `array[size] = value;` since `size` should represent the number of elements in the list, not the next available index.